weight.cpp

```diff
@@ -42,6 +43,33 @@ bool is_active(const weight &w)
  return w.is_active();
 }
 
+std::vector<double> convert_to_double
+  (const std::vector<weight> &weights)
```

Comment:
Still ok in terms of line length
```suggestion
std::vector<double> convert_to_double (const std::vector<weight> &weights)
```